docs: update readme

ltfschoen opened the pull request.

Tbaut commented:
These are very specific changes to your needs. There is no need for changing the name, or install nvm.

ltfschoen commented:
> These are very specific changes to your needs. There is no need for changing the name, or install nvm.

i've updated it so use of nvm is optional and removed other stuff that isn't essential

TarikGul commented on README.md:
```diff
@@ -18,19 +18,23 @@ Find out more about how to use the extension as a Dapp developper, cookbook, as
-    - go to `about:debugging#addons`
-    - check "Enable add-on debugging"
-    - click on "Load Temporary Add-on" and point to `packages/extension/build/manifest.json`
+    - go to "This Firefox" at `about:debugging#addons`
```
What does `"This Firefox"` mean?

ltfschoen replied:
it's a menu item in latest Firefox 124.0.1 as shown below. if i go to `about:debugging#addons` it redirects me to `about:debugging#/runtime/this-firefox`

<img width="986" alt="Screenshot 2024-03-25 at 8 01 44 am" src="https://github.com/polkadot-js/extension/assets/6226175/a8193483-b658-4849-9055-974a82585e87">

Tbaut replied:
This is the default tab when going to `about:debugging#addons`, so I suggest to remove this mention as it may change.
```suggestion
    - go to `about:debugging#addons`
```

Tbaut commented on README.md:
```diff
@@ -18,19 +18,23 @@ Find out more about how to use the extension as a Dapp developper, cookbook, as
-    - check "Enable add-on debugging"
-    - click on "Load Temporary Add-on" and point to `packages/extension/build/manifest.json`
+    - go to "This Firefox" at `about:debugging#addons`
+    - check "Enable add-on debugging" (if available)
```
this is not present in new versions, let's remove it then.
```suggestion
```

Tbaut commented on README.md:
```diff
@@ -18,19 +18,23 @@ Find out more about how to use the extension as a Dapp developper, cookbook, as
     - if developing, after making changes - reload the extension
-3. When visiting `https://polkadot.js.org/apps/` it will inject the extension
+    - edit the extension settings to "Run in Private Windows" if necessary
+    - click the puzzle icon, then click the gear icon next to the "polkadot{.js} extension (development only)" extension and choose "Pin to Toolbar"
```
This is really not needed either and can change in the future. I'd remove.

```suggestion
```
